### PR TITLE
Add logging for spiderlings spawned by botany + refactor spawning code

### PR DESCRIPTION
**Pull request by optimumtact**
Adds a log with an admin message for when spiderlings are spawned by a
botany user adding mutagen or adminordrazine, it also adds a helper
method for spawning atoms to handle a little bit of the copypaste where it can

### CODE REVIEW

File: code/__HELPERS/mobs.dm [hunk 1]
```diff
@@ -330,6 +330,18 @@ Proc for attack log creation, because really why not
 		if(H.dna && istype(H.dna.species, species_datum))
 			. = TRUE
 
+/proc/spawn_atom(spawn_type, target, amount, admin_spawn=FALSE, callback_src=null, callback_method=null)
+	var/turf/T = get_turf(target)
+	if(!T)
+		throw EXCEPTION("attempt to spawn atom type: [spawn_type] in nullspace")
+
+	for(var/j in 1 to amount)
+		var/atom/movable/X = new spawn_type(T)
```
Comment (optimumtact) [inline code/__HELPERS/mobs.dm]: herp that should be a var/atom

File: code/modules/hydroponics/hydroponics.dm [hunk 2]
```diff
@@ -440,16 +440,21 @@
 		dead = 1
 
 
-/obj/machinery/hydroponics/proc/mutatepest()
+
+/obj/machinery/hydroponics/proc/mutatepest(mob/user)
 	if(pestlevel > 5)
+		message_admins("[ADMIN_LOKUPFLW] caused spiderling pests to spawn in a hydro tray")
```
Comment (phil235) [inline code/modules/hydroponics/hydroponics.dm]: ADMIN_LOOKUPFLW

File: code/modules/hydroponics/hydroponics.dm [hunk 3]
```diff
@@ -440,16 +440,21 @@
...
+
+/obj/machinery/hydroponics/proc/mutatepest(mob/user)
 	if(pestlevel > 5)
+		message_admins("[ADMIN_LOKUPFLW] caused spiderling pests to spawn in a hydro tray")
+		log_game("[key_name(user)] caused spiderling pests to spawn in a hydro tray")
 		visible_message("<span class='warning'>The pests seem to behave oddly...</span>")
-		for(var/i=0, i<3, i++)
-			var/obj/structure/spider/spiderling/S = new(src.loc)
-			S.grow_as = /mob/living/simple_animal/hostile/poison/giant_spider/hunter
+		spawn_atom(/obj/structure/spider/spiderling, src, 3, src, "post_spiderling_spawn")
```
Comment (ChangelingRain) [inline code/modules/hydroponics/hydroponics.dm]: admin_spawn arg being passed src? I think you need to either name your arguments or give it FALSE for admin_spawn

Comment (optimumtact) [inline code/modules/hydroponics/hydroponics.dm]: whoops, I added teh admin_spawn as an afterthought, I'll change that

File: code/modules/hydroponics/hydroponics.dm (hunk 2 above)
Comment (lzimann) [inline code/modules/hydroponics/hydroponics.dm]: ADMIN_LOOKUPFLW(user)

File: code/modules/hydroponics/hydroponics.dm [hunk 4]
```diff
@@ -440,16 +440,21 @@
...
 	else
-		usr << "<span class='warning'>The pests seem to behave oddly, but quickly settle down...</span>"
+		user << "<span class='warning'>The pests seem to behave oddly, but quickly settle down...</span>"
+
+/obj/machinery/hydroponics/proc/post_spiderling_spawn(/obj/structure/spider/spiderling/spider)
+	if(spider && !qdeleted(spider) && istype(spider))
+		spider.grow_as = /mob/living/simple_animal/hostile/poison/giant_spider/hunter
```
Comment (phil235) [inline code/modules/hydroponics/hydroponics.dm]: so we're calling two additional procs, and there's more lines in the code. Why is it better than the previous code?

Comment (phil235) [inline code/modules/hydroponics/hydroponics.dm]: How is that cleaner? You have to give a new proc to every object that uses spawn_atom

Comment (ChangelingRain) [inline code/modules/hydroponics/hydroponics.dm]: yeah this is really fucky, honestly, it's a lot more lines for nothing good

**Comment by PKPenguin321** [comment]
cool now please move AI law change logging to the investigate verb instead of the secrets panel

File: code/__HELPERS/mobs.dm [hunk 5]
```diff
@@ -330,6 +330,15 @@ Proc for attack log creation, because really why not
 		if(H.dna && istype(H.dna.species, species_datum))
 			. = TRUE
 
+/proc/spawn_atom(spawn_type, target, amount, admin_spawn=FALSE)
+	var/turf/T = get_turf(target)
```
Comment (ChangelingRain) [inline code/__HELPERS/mobs.dm]: You should let this spawn in the target instead of only on the target's turf.

File: code/__HELPERS/mobs.dm [hunk 6]
```diff
@@ -330,6 +330,15 @@ Proc for attack log creation, because really why not
 		if(H.dna && istype(H.dna.species, species_datum))
 			. = TRUE
 
+/proc/spawn_atom_to_turf(spawn_type, target, amount, admin_spawn=FALSE)
+	var/turf/T = get_turf(target)
```
Comment (ChangelingRain) [inline code/__HELPERS/mobs.dm]: good sir why is this restricted to turfs, good sir,

Comment (optimumtact) [inline code/__HELPERS/mobs.dm]: pulls to my branch welcome

**Pull request merged**